### PR TITLE
Social Urls: move network code to util

### PR DESCRIPTION
**Pull request by jrefano**

### CODE REVIEW

File: util/socialUrl.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+define(function() {
+  'use strict';
+
+  return function(services, url, service) {
+    url = url.trim();
+    url = url.replace(/https?:\/\//, '');
```
Comment (mikesherov) [inline util/socialUrl.js]: this line isn't being tested.

File: util/socialUrl.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+define(function() {
+  'use strict';
+
+  return function(services, url, service) {
+    url = url.trim();
```
Comment (mikesherov) [inline util/socialUrl.js]: this line isn't being tested.

File: util/socialUrl.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    url = url.replace(/https?:\/\//, '');
+    url = url.replace('www.', '');
```
Comment (mikesherov) [inline util/socialUrl.js]: this line isn't being tested.

File: test/specs/util/socialUrl.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
+define(['util/socialUrl'], function(socialUrl) {
```
Comment (mrjoelkemp) [inline test/specs/util/socialUrl.js]: Non-blocker: the naming of this module isn't representative of what it's doing. It's more of a socialUrlValidator

Comment (mrjoelkemp) [inline test/specs/util/socialUrl.js]: even `socialUrlCleaner` would be a better name

Comment (jrefano) [inline test/specs/util/socialUrl.js]: true

File: util/socialUrl.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    Object.keys(services).forEach(function(key) {
+      var domain = services[key].domain;
+      var prefix = services[key].prefix;
+
+      if (!(service in services) || key === service) {
+        url = url.replace(prefix, '');
```
Comment (mrjoelkemp) [inline util/socialUrl.js]: Non-blocker: Would you be able to do: url.replace(new RegExp('[' + prefix + '|' + domain + ']'), '')?

**Comment by mrjoelkemp** [comment]
:+1:

File: test/specs/util/socialUrlCleaner.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    it('can clean https url', function() {
+      expect(socialUrl(this._urls, 'https://foo.com')).toEqual('foo.com');
+    });
+
+    it('removes www from url', function() {
+      expect(socialUrl(this._urls, 'www.foo.com')).toEqual('foo.com');
+    });
+
+    it('removes www and https from url', function() {
+      expect(socialUrl(this._urls, 'https://www.foo.com')).toEqual('foo.com');
```
Comment (mikesherov) [inline test/specs/util/socialUrlCleaner.js]: One test for the whole shebang please:

```
expect(socialUrl(this._urls, '   https://www.google.com/+foo       ')).toEqual('foo');
```

**Comment by mikesherov** [comment]
:+1:

**Pull request merged**